src/Raven.Server/Documents/Replication/IncomingReplicationHandler.cs

```diff
@@ -341,11 +348,31 @@ void HandleSocketException(SocketException e)
                 if (_cts.IsCancellationRequested)
                     return;
 
+                DynamicJsonValue returnValue;
+
+                if (e is AggregateException ag && ag.InnerExceptions.Count == 1 && ag.InnerExceptions[0] is MissingAttachmentException mae)
```

Comment:
You can use `ExtractSingleInnerException` to simplify that

Reply:
okay